docs/usage/language-constraints-upgrading.md

```diff
@@ -0,0 +1,87 @@
+
+## Restricting upgrades to compatible releases
+
+By default Renovate _does not_ apply language constraints to upgrades, which means it will propose "any" stable upgrade without consideration of language version.
+For certain ecosystems, changes to language constraints are usually done in major releases and documented in the release notes, so this behavior may not be problematic.
+For other ecosystems this may seem simply _wrong_.
+
+Renovate users can opt into strict compatibility filtering using `constraintsFiltering=strict`.
+You should understand the limitations of this first, also as an understanding of why it's not the default behavior.
+
+## Language constraint updating
+
+The first challenge is that Renovate may not yet support the ability to update your language constraints in an automated manner, and even when it does, users may not understand how many updates are depending on it.
+
+For example, consider a Node.js project which has its `engines` set to `"node": "^18.0.0 || ^20.0.0"`.
+
+Should Renovate skip Node.js v21 because it's non-LTS?
+When Node.js v22 comes out, should Renovate add it to your engines, or wait until it starts LTS?
+When Node.js v18 is EOL, should Renovate drop it from the engines list?
+
+It's hard for Renovate to know or guess what users want, because users have strong but also different opinions on the answer to the above questions.
+
+Additionally, even if Renovate guesses right or adds advanced capabilities to allow this to be configurable, then users might sit any of these "major" upgrades for months.
+If a project delays creating or merging the update to drop Node.js v18 from engines then it won't be able to upgrade to any new versions of library dependencies which themselves already dropped support.
+
+## Strict filtering limitations
+
+Consider again the Node.js project which has its `engines` set to `"node": "^18.0.0 || ^20.0.0"`.
+
+Now also consider a library which sets its `engines` to `"node": "^18.12.0 || ^20.9.0"` because it wants to only support "LTS releases" of Node.js.
+Strictly speaking, this dependency is not compatible with the project above which has wider requirements for Node versions, and Renovate would hold back any upgrades for it.
+Should Renovate somehow reason and _assume_ that this narrower engines support is actually OK?
+What if the project _already_ used an existing version of this library "incorrectly"?
+
+A second problem is that if Renovate is unable to update the language constraints, or the user ignores or does not notice the language upgrade, then they also may not realize that many of their dependencies are out of date because Renovate is not proposing PRs.
+For example, a project may have 10 dependencies, and 8 of those have updates but all 8 require the project to update its language constraints first.
+The project administrator may think they are mostly up to date, because Renovate is not proposing PRs, but in reality 80% of their dependencies are outdated.
```

Comment:
```suggestion
The project administrator thinks they are up to date, because Renovate is not creating PRs, but 80% of their dependencies are outdated.
```